client/actions/shield-study/index.js

```diff
@@ -6,8 +6,34 @@ export default class ShieldStudyAction extends Action {
     this.storage = normandy.createStorage(recipe.id);
   }
 
+  setHasBeenShown() {
+    this.storage.setItem('studyHasBeenShown', true);
+  }
+
+  async getHasBeenShown() {
+    const hasBeenShown = await this.storage.getItem('studyHasBeenShown');
+    return hasBeenShown;
+  }
+
   async execute() {
+    const storageIsDurable = await this.storage.isDurable();
+
+    if (!storageIsDurable && !this.normandy.testing) {
+      throw new Error('Storage durability unconfirmed');
```

Comment:
Good call. My only problem with that would be that I'm using `this.getItem` in the `_setDurabilityStatus` method. So if `getItem` returned `null` or rejected, we would never actually be able to increment the durability status value.

Any thoughts on getting around that? I could add a check in `getItem` that wouldn't reject if we were getting the `storageDurability` key?